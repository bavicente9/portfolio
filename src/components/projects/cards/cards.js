import './cards.css'

//return each card with its info according to the json
const generateCards = (page, infoJson) => {
    let cardsComponents = []
    for (let i = (page * 3 - 3); i < (page * 3); i++) {
        const item = infoJson[i]
        //check if there is not more items
        if (typeof item === 'undefined') break

        cardsComponents.push(
            <div key={i} className='projects__card'>
                <div className='projects__cardImage'>
                    <img alt='' src={item.image} />
                </div>
                <div className='projects__cardInfo'>
                    <h2>{item.name}</h2>
                    <ul className='projects__cardInfoTechnologies'>
                        {item.technologies.map((tech, index) => <li key={index}>{tech}</li>)}
                    </ul>
                    <p>{item.description}</p>
                    <nav className='projects__cardButtons' aria-label='project links'>

                        {item.link_liveDemo ==='undefined'?null:<a rel='external' className='button' href={item.link_liveDemo} target="_blank">Live Demo</a>}
                        {item.link_github ==='undefined'?null:<a rel='external' className='button button--white' href={item.link_github} target="_blank">Github</a>}
                    </nav>
                </div>
            </div>
        )
    }
    return (<>{cardsComponents}</>)

}

export const Cards = ({page, infoJson})=>{
    return(<>{generateCards(page, infoJson)}</> )
}