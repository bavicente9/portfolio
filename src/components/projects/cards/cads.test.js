import {render} from '@testing-library/react'
import { Cards } from './cards'

describe('Testing Projects', ()=>{
    const infoJson = JSON.parse(`[{
        "name": "project Name5",
        "image": "./images/maribelnails.png",
        "technologies": [
            "React",
            "JavaScript",
            "Css",
            "HTML"
        ],
        "description": "Here goes the project's description: exampmle: Netflix clone made using React and TMDB API. This app fetches the movies data from TMDB database. I have used plain CSS to style the components and make it look like the original Netflix website",
        "link_liveDemo": "https://github.com",
        "link_github": "https://github.com"
    }]`)

    it('renders correctly', ()=>{
        const {asFragment} = render(<Cards page={1} infoJson={infoJson} />)
        expect(asFragment()).toMatchSnapshot()
    })



})