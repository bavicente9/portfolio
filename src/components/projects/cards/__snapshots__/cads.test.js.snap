// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Projects renders correctly 1`] = `
<DocumentFragment>
  <div
    class="projects__card"
  >
    <div
      class="projects__cardImage"
    >
      <img
        alt=""
        src="./images/maribelnails.png"
      />
    </div>
    <div
      class="projects__cardInfo"
    >
      <h2>
        project Name5
      </h2>
      <ul
        class="projects__cardInfoTechnologies"
      >
        <li>
          React
        </li>
        <li>
          JavaScript
        </li>
        <li>
          Css
        </li>
        <li>
          HTML
        </li>
      </ul>
      <p>
        Here goes the project's description: exampmle: Netflix clone made using React and TMDB API. This app fetches the movies data from TMDB database. I have used plain CSS to style the components and make it look like the original Netflix website
      </p>
      <nav
        aria-label="project links"
        class="projects__cardButtons"
      >
        <a
          class="button"
          href="https://github.com"
          rel="external"
          target="_blank"
        >
          Live Demo
        </a>
        <a
          class="button button--white"
          href="https://github.com"
          rel="external"
          target="_blank"
        >
          Github
        </a>
      </nav>
    </div>
  </div>
</DocumentFragment>
`;
